experimental/HttpBenchmark/Program.cs

```diff
@@ -0,0 +1,384 @@
+#endif
+
+            var gc = GCSettings.IsServerGC ? "server" : "client";
+            Console.WriteLine($"GC: {gc}");
+
+            Console.WriteLine($"Connections: {connections}");
+            Console.WriteLine($"Duration: {duration}");
+            Console.WriteLine($"Pipeline: {pipeline}");
+            Console.WriteLine($"Keepalive: {keepalive}");
+            Console.WriteLine($"Uri: {uri}");
+
+            Console.WriteLine();
+
+#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+            WriteResults(duration);
+#pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+        }
+
+        private static async Task<Socket> CreateSocket(Uri uri)
+        {
+            var tcs = new TaskCompletionSource<Socket>();
+
+            var socketArgs = new SocketAsyncEventArgs();
+            socketArgs.RemoteEndPoint = new DnsEndPoint(uri.DnsSafeHost, uri.Port);
+            socketArgs.Completed += (s, e) => tcs.TrySetResult(e.ConnectSocket);
+
+            // Must use static ConnectAsync(), since instance Connect() does not support DNS names on OSX/Linux.
+            if (Socket.ConnectAsync(SocketType.Stream, ProtocolType.Tcp, socketArgs))
+            {
+                await tcs.Task;
+            }
+
+            var socket = socketArgs.ConnectSocket;
+
+            if (socket == null)
+            {
+                throw new SocketException((int)socketArgs.SocketError);
+            }
+            else
+            {
+                return socket;
+            }
+        }
+
+        private static void SendReceive(Socket socket, byte[] requestBytes, int responseLength, byte[] responseBuffer)
+        {
+            socket.Send(requestBytes);
+
+            int bytesReceived = 0;
+            while (bytesReceived < responseLength)
+            {
+                bytesReceived += socket.Receive(responseBuffer, bytesReceived, responseBuffer.Length - bytesReceived, SocketFlags.None);
+            }
+        }
+
+        private static async Task WriteResults(TimeSpan duration)
+        {
+            var lastRequests = (long)0;
+            var lastElapsed = TimeSpan.Zero;
+
+            while (true)
+            {
+                await Task.Delay(TimeSpan.FromSeconds(1));
+
+                var currentRequests = _requests - lastRequests;
+                lastRequests = _requests;
+
+                var elapsed = _stopwatch.Elapsed;
+                var currentElapsed = elapsed - lastElapsed;
+                lastElapsed = elapsed;
+
+                WriteResult(_requests, currentRequests, currentElapsed);
+
+                if (elapsed > duration)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine($"Average RPS: {Math.Round(_requests / elapsed.TotalSeconds)}");
+
+                    Environment.Exit(0);
```

Comment:
Maybe I'm just being too much of a purist, but I kinda feel like this is a bit of a weird way to shut down the benchmark. It took me a bit to realize the benchmark finished here. I guess I'd kinda prefer something like a timed cancellation token or something that collected the results after the run... but I also see the value in collecting as we go.

I guess I'm OK with this model, but could we put some comments in the benchmark loops themselves indicating how they are terminated? `while(true)` always makes me nervous, especially if I can't see the `break` anywhere in the loop :)